packages/server/api.js

```diff
@@ -234,6 +234,56 @@ module.exports = async function attachAPI(app, {wss, db, dbDir}) {
     ])
   }
 
+  app.post('/api/emotes', [
+    ...middleware.loadSessionID('sessionID'),
+    ...middleware.getSessionUserFromID('sessionID', 'sessionUser'),
+    ...middleware.requireBeAdmin('sessionUser'),
+    ...middleware.loadVarFromBody('shortcode'),
+    ...middleware.validateVar('shortcode', validate.string),
+    ...middleware.requireNameValid('shortcode'),
+    ...middleware.loadVarFromBody('imageURL'),
+    ...middleware.validateVar('imageURL', validate.string),
+
+    async function(request, response, next) {
+      const { imageURL, shortcode } = request[middleware.vars]
+
+      if (await db.emotes.findOne({shortcode})) {
+        response.status(400).json(errors.NAME_ALREADY_TAKEN)
+        return
+      }
+
+      const newEmote = {imageURL, shortcode}
+      await db.emotes.insert(newEmote)
```

Comment:
Since it's unique anyway it might be worth using `_id` rather than `shortcode` on the db side. Small optimization, I guess.

Reply:
Naw, I agree but this future-proofs it from any wild ideas we have in the future where we might want two emotes to have the same shortcode. No need to actively *prevent* that. What's going to take up *more* database space are individual messages themselves :)